src/test/java/uk/gov/hmcts/darts/transcriptions/validator/TranscriptionApproveMarkForDeletionValidatorTest.java

```diff
@@ -0,0 +1,158 @@
+    }
+
+    @Test
+    void validate_shouldThrowException_whenHiddenReasonNotMarkedForDeletion() {
+        ObjectAdminActionEntity entity = new ObjectAdminActionEntity();
+        ObjectHiddenReasonEntity hiddenReason = new ObjectHiddenReasonEntity();
+        hiddenReason.setMarkedForDeletion(false);
+        entity.setObjectHiddenReason(hiddenReason);
+        when(objectAdminActionRepository.findByTranscriptionDocument_Id(TRANSCRIPTION_DOCUMENT_ID))
+            .thenReturn(List.of(entity));
+        when(objectHiddenReasonRepository.findById(any())).thenReturn(Optional.of(hiddenReason));
+
+        DartsApiException exception = assertThrows(DartsApiException.class,
+                                                   () -> validator.validate(TRANSCRIPTION_DOCUMENT_ID));
+        assertEquals(TranscriptionApiError.TRANSCRIPTION_DOCUMENT_MARKED_FOR_DELETION_REASON_NOT_FOUND, exception.getError());
+    }
+
+    @Test
+    void validate_shouldThrowException_whenApprovedByHiddenByUser() {
+        ObjectAdminActionEntity entity = new ObjectAdminActionEntity();
+        ObjectHiddenReasonEntity hiddenReason = new ObjectHiddenReasonEntity();
+        hiddenReason.setMarkedForDeletion(true);
+        entity.setObjectHiddenReason(hiddenReason);
+        UserAccountEntity user = new UserAccountEntity();
+        user.setId(1);
+        entity.setHiddenBy(user);
+        when(objectAdminActionRepository.findByTranscriptionDocument_Id(TRANSCRIPTION_DOCUMENT_ID))
+            .thenReturn(List.of(entity));
+        when(objectHiddenReasonRepository.findById(any())).thenReturn(Optional.of(hiddenReason));
+        when(userIdentity.getUserAccount()).thenReturn(user);
+
+        DartsApiException exception = assertThrows(DartsApiException.class,
+                                                   () -> validator.validate(TRANSCRIPTION_DOCUMENT_ID));
+        assertEquals(TranscriptionApiError.TRANSCRIPTION_DOCUMENT_DELETION_CAN_NOT_APPROVE_OWN_REQUEST, exception.getError());
+    }
+
+    @Test
+    void validate_shouldNotThrowException_whenAllConditionsAreMet() {
+        ObjectAdminActionEntity entity = new ObjectAdminActionEntity();
+        ObjectHiddenReasonEntity hiddenReason = new ObjectHiddenReasonEntity();
+        hiddenReason.setMarkedForDeletion(true);
+        entity.setObjectHiddenReason(hiddenReason);
+        UserAccountEntity hiddenByUser = new UserAccountEntity();
+        hiddenByUser.setId(1);
+        entity.setHiddenBy(hiddenByUser);
+        UserAccountEntity currentUser = new UserAccountEntity();
+        currentUser.setId(2);
+        when(objectAdminActionRepository.findByTranscriptionDocument_Id(TRANSCRIPTION_DOCUMENT_ID))
+            .thenReturn(List.of(entity));
+        when(objectHiddenReasonRepository.findById(any())).thenReturn(Optional.of(hiddenReason));
+        when(userIdentity.getUserAccount()).thenReturn(currentUser);
+
+        assertDoesNotThrow(() -> validator.validate(TRANSCRIPTION_DOCUMENT_ID));
```

Comment:
We should also add relevant verify(...) here to ensure all the right things are being hit

Reply:
Done